components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -1507,25 +1507,51 @@ def getRetvalDeclarationForType(returnType, descriptorProvider):
                     returnType)
 
 
-def MemberCondition(pref, func):
+def MemberCondition(pref, func, exposed):
     """
     A string representing the condition for a member to actually be exposed.
     Any of the arguments can be None. If not None, they should have the
     following types:
 
     pref: The name of the preference.
     func: The name of the function.
+    exposed: One or more names of an exposed member.
     """
     assert pref is None or isinstance(pref, str)
     assert func is None or isinstance(func, str)
-    assert func is None or pref is None
+    assert exposed is None or isinstance(exposed, str) or isinstance(exposed, list)
+    assert func is None or pref is None or exposed is None
     if pref:
         return 'Condition::Pref("%s")' % pref
     if func:
         return 'Condition::Func(%s)' % func
+    if exposed:
+        if isinstance(exposed, str):
+            return "Condition::Exposed(%s)" % ExposedCondition(exposed)
     return "Condition::Satisfied"
 
 
+def ExposedCondition(cond):
+    exposed_dict = dict({
+        "Window": "Window",
+        "Worklet": "PaintWorkletGlobalScope",
+        "Worker": "DedicatedWorkerGlobalScope",
+        "ServiceWorker": "ServiceWorkerGlobalScope",
+        "DedicatedWorker": "DedicatedWorkerGlobalScope",
+        "PaintWorklet": "PaintWorkletGlobalScope",
+        "TestWorklet": "TestWorkletGlobalScope"
+    })
```

Comment:
We should be able to generate this dictionary programmatically:
* `global_descriptors = config.getDescriptors(isGlobal=True)` gives us all of the interfaces that have a `Global` annotation
* `descriptor.interface.getExtendedAttribute("Global")` gives us the list of values for a particular interface